instance/models/openedx_instance.py

```diff
@@ -415,12 +410,18 @@ def terminate_obsolete_appservers(self, days=2):
     def shut_down(self):
         """
         Shut down this instance.
-
-        This process consists of two steps:
-
-        1) Disable New Relic monitors.
-        2) Terminate all app servers belonging to this instance.
         """
-        self.disable_monitoring()
-        for appserver in self.appserver_set.all():
+        if self.load_balancing_server is not None:
+            load_balancer = self.load_balancing_server
+            self.load_balancing_server = None
+            self.save()
+            if self.active_appserver is None:
+                # If an appserver is active, reconfiguring the load_balancer happens
+                # implicitly when terminate_vm() is called further down.
+                load_balancer.reconfigure()
+        self.remove_dns_records()
+        self.deprovision_mysql()
+        self.deprovision_mongo()
+        self.deprovision_swift()
```

Comment:
There is no reason for this order, and doing it the other way around sounds good.

However, I don't understand your comment about "keeping the previous order".  The old version did not call the `deprovision_*()` methods at all when calling `shut_down()`, but only when the instance was deleted.  The `delete()` and `shut_down()` methods both had clean-up code that was mostly disjoint, and I consolidated that code.  I'm not sure we actually _want_ to deprovision the databases when calling `shut_down()`, but I think we do.